### PR TITLE
Run pipeline for binary files on pull if go code changed

### PR DESCRIPTION
**Pull request by 6543**
will resolve a todo on #684

### CODE REVIEW

File: .woodpecker/docker.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+depends_on:
```
Comment (6543) [inline .woodpecker/docker.yml:1]: formation nit

File: .woodpecker/binaries.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
+depends_on:
+  - test
+  - web
+
+pipeline:
```
Comment (6543) [inline .woodpecker/binaries.yml:5]: is 1:1 the `main.yml` just added path condition to each step

**Review by anbraten — COMMENTED** [review]
Instead of `when.path.include` we should be able to use `when.path` directly. #86 will be super helpful such configs at some point. 😂

File: .woodpecker/binaries.yml [hunk 3]
```diff
@@ -18,6 +18,8 @@ pipeline:
           # go source code
           - "**/*.go"
           - "go.*"
+          # web source code
+          - "web/"
```
Comment (anbraten) [inline .woodpecker/binaries.yml:22]: Sorry, I think we have to adjust this to `web/**`. I tried it by adding a new test case in `pipeline/frontend/yaml/constraint/constraint_test.go` which was throwing an error:

```go
		{
			conf: "'docs/'",
			with: []string{"docs/README.md", "docs/sub/README.md", "docs/sub-sub/README.md"},
			want: true,
		},
```

Comment (6543) [reply .woodpecker/binaries.yml:22]: Its blob pattern ...

I'll update it ...

**Comment by 6543** [comment]
PS: remove `include:` result in jsonschema report an error - I think we have to fix that ... but it's another pull to do s

**Review by anbraten — APPROVED** [review]

**Pull request merged**